File: algo_go/which_power.go
package main

import "fmt"

const MAX_LEN = 100000

func main() {
    var school [MAX_LEN]int;
    var n, schid, score int;
    fmt.Scanf("%d", &n);
    for i := 0; i < n; i++ {
        fmt.Scanf("%d %d", &schid, &score);
        school[schid] += score;
    }

    k := 0;
    MAX := -1;
    for i := 0; i < n; i++ {
        if (school[i] > MAX) {
            MAX = school[i];
            k = i;
        }
    }
    fmt.Printf("%d %d\n", k, MAX);
}

File: algo-c/oldfiles/which_power.c
#include <stdio.h>

#define MAX_LEN 100000
//int school[MAX_LEN] = {0};

int main(void) {
    // const int MAX_LEN = 100000;
    int school[MAX_LEN] = {0};
    int n, schid, score;

    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%d%d", &schid, &score);
        school[schid] += score;
    }

    int k = 0, MAX = -1;
    for (int i = 0; i < n; i++) {
        if (school[i] > MAX) {
            MAX = school[i];
            k = i;
        }
    }

    printf("%d %d\n", k, MAX);
    return 0;
}
